metricbeat/schema/error.go

```diff
@@ -0,0 +1,34 @@
+package schema
+
+import "fmt"
+
+const (
+	RequiredType = iota
+	OptionalType = iota
+)
+
+type Error struct {
+	key       string
+	message   string
+	errorType int
```

Comment:
Would it make sense to make ErrorType a type so that it's restricted to the types above?